### DISCUSSION TITLE
GraphQL support via APQ

### DISCUSSION THREAD

**Discussion by stafyniaksacha** [strapi-community/strapi-plugin-rest-cache#17] [Ideas]
This plugin aim to speed up HTTP requests with server side cache.
We can not cache POST requests unless we implement the corresponding [RFC 2616](http://www.faqs.org/rfcs/rfc2616.html).

The idea would be to manage Automatic Persisted Queries (AQP) via this plugin.
https://www.apollographql.com/docs/apollo-server/performance/apq/

My questions are:
- what would be the configuration for this?
- how will this not conflict with rest content?
- what can be the tests cases?
- Do someone want to lead this topic?

**Comment by gvocale** [comment]
I have turned on Automatic Persisted Queries in my next JS app, fetching from Strapi via GraphQL.

The graphql queries are being fetched in GET and seem to work correclty, but when I inspect my redis cache I see no key saved in it.

Is the redis cache being used when querying via GraphQL?

**Reply by stafyniaksacha** [reply]
No, nothing has been made for GraphQL on this plugin for now. We need to find a way to identify requests to be able to purge cache when the content get updated to not have stalled data.

**Reply by derrickmehaffy** [reply]
I'm honestly of the opinion that a separate plugin should be built for GraphQL but it could utilize the same providers. The logic to handle REST vs GraphQL are very different and if we try to build in cache support into a single plugin for both it would mean including dependencies that apply to both.

Since the Strapi GraphQL plugin is optional I think having the cache for GraphQL also should be optional when compared to REST + REST Caching. Also the name of the plugin is `strapi-plugin-rest-cache` so we hit the challenge of renaming the plugin as well.
